src/util/src/tskiplist.c

```diff
@@ -124,28 +129,33 @@ SSkipListNode *tSkipListPut(SSkipList *pSkipList, void *pData) {
   return pNode;
 }
 
-// Put a batch of data into skiplist. The batch of data must be in ascending order
-void tSkipListPutBatch(SSkipList *pSkipList, void **ppData, int ndata) {
+void tSkipListPutBatchByIter(SSkipList *pSkipList, void *iter, iter_next_fn_t iterate, void **lastData) {
   SSkipListNode *backward[MAX_SKIP_LIST_LEVEL] = {0};
   SSkipListNode *forward[MAX_SKIP_LIST_LEVEL] = {0};
   bool           hasDup = false;
   char *         pKey = NULL;
   char *         pDataKey = NULL;
   int            compare = 0;
+  SSkipListNode* pNode;
 
   tSkipListWLock(pSkipList);
 
+  void* pData = iterate(iter);
+  if(pData == NULL) return;
+
   // backward to put the first data
-  hasDup = tSkipListGetPosToPut(pSkipList, backward, ppData[0]);
-  tSkipListPutImpl(pSkipList, ppData[0], backward, false, hasDup);
+  hasDup = tSkipListGetPosToPut(pSkipList, backward, pData);
+
+  pNode = tSkipListPutImpl(pSkipList, pData, backward, false, hasDup);
+  if(pNode) *lastData = pNode->pData;
```

Comment:
The lastData assignment logic should not be in skiplist. It should be in app code. So just put the logic in the callback function.